src/shadowbox/server/shared_metrics.ts

```diff
@@ -156,6 +158,7 @@ export class OutlineSharedMetricsPublisher implements SharedMetricsPublisher {
   // metricsUrl: where to post the metrics
   constructor(
       private clock: Clock, private serverConfig: JsonConfig<ServerConfigJson>,
+      private keyConfig: JsonConfig<AccessKeyConfigJson>,
```

Comment:
Why not pass the access key repository instead of the backing file?

Reply:
A couple reasons:

First, the access key repo has no way to get a single key without copying the whole list of keys without adding functionality.  Passing the config allows us to directly access the keys, and the config class is even written with the metrics use case in mind.  

Second, we don't need any of the other functionality of the access key repo than just listing the keys, which we get for free with the key config.

Third, stylistic parity with the act that we already pass the server config.

Reply:
Let's not expose the serialization format. The server config was being passed because that was a configuration for the publisher. It's a different case.

Let's pass the repo instead as Alberto suggested. Why do you need to get a single key? Don't you need to go over all of them anyway?

Reply:
Or perhaps the feature metrics could be added to UsageMetrics. Or create a FeatureMetrics object like that one.